### ISSUE TITLE
Pod with large memory failed/slow to start

### ISSUE THREAD

**Issue by liyimeng** [kata-containers/kata-containers#9351]
# Get your issue reviewed faster

To help us understand the problem more quickly, please do the following:

# Description of problem

I try to start pod with 128G memory request with k8s. It spin for a while and failed，however qemu process is actually running.  
If I reduce the pods memory request to 98G, it works without problem. 

# Expected result

Pod with large memory request should be started

# Actual result

It is not, or at least not in healthy start. And the containerd are flooding with these log
```
time="2024-03-26T18:35:23.503116822+08:00" level=warning msg="Large hotplug. Adding 98304 MB of 720564 total memory" name=containerd-shim-v2 pid=31717 sandbox=0253d9313280886abab053a6e3510c7212b6fbdc7546dab9d7b0df4c14e31a5e source=virtcontainers subsystem=sandbox
time="2024-03-26T18:35:23.502409602+08:00" level=warning msg="Large hotplug. Adding 98304 MB of 865087 total memory" name=containerd-shim-v2 pid=31717 sandbox=0253d9313280886abab053a6e3510c7212b6fbdc7546dab9d7b0df4c14e31a5e source=virtcontainers subsystem=sandbox
time="2024-03-26T18:35:23.503164328+08:00" level=warning msg="Large hotplug. Adding 98304 MB of 720564 total memory" name=containerd-shim-v2 pid=31717 sandbox=0253d9313280886abab053a6e3510c7212b6fbdc7546dab9d7b0df4c14e31a5e source=virtcontainers subsystem=sandbox
time="2024-03-26T18:35:23.503239598+08:00" level=warning msg="Large hotplug. Adding 98304 MB of 720564 total memory" name=containerd-shim-v2 pid=31717 sandbox=0253d9313280886abab053a6e3510c7212b6fbdc7546dab9d7b0df4c14e31a5e source=virtcontainers subsystem=sandbox

```

# Further information

my node have 256G physical ram in total. 

# Kata Containers survey

Please consider taking the survey to help us help you: https://openinfrafoundation.formstack.com/forms/kata_containers_user_survey

**Comment by lifupan** [comment]
Hi @liyimeng 

Since the big memory will take a lot of time to do the memory hotplug. 
If you want to make it boot quickly, you can set the default_memory as a bigger one, this could
reduce the hotplug time and make your pod start quickly.

**Comment by liyimeng** [comment]
@lifupan aha, good to know. So I have to balance in some way. My understanding is that if I set default_memory too large, small pod will consume unnecessary memory, do I get it right in this part?

Thanks for kindly help!

**Comment by lifupan** [comment]
> @lifupan aha, good to know. So I have to balance in some way. My understanding is that if I set default_memory too large, small pod will consume unnecessary memory, do I get it right in this part?
> 
> Thanks for kindly help!

Yes, you are right. Of course, you can set the default memory using annotation, so that you can specify only one pod instead of affected the others. such as specify the annotation io.katacontainers.config.hypervisor.default_memory = <memory size>

**Comment by liyimeng** [comment]
Thanks @lifupan
